public/jsx/views/card-listing.jsx

```diff
@@ -23,25 +23,16 @@ module.exports = React.createClass({
   },
 
   componentDidMount: function() {
-    $.ajax({
-      method: 'get',
-      url: '/api/braintree/mozilla/paymethod/',
-      context: this,
-    }).then(function(data) {
-      if (this.isMounted()) {
-        // Ignore react/no-did-mount-set-state eslint warning
-        if (data.length) {
-          console.log('Card data found, showing card form');
-          this.setState({cards: data}); // eslint-disable-line
-        } else {
-          console.log('No card data found, showing card form');
-          this.transitionTo('card-form');
-        }
-      }
-    }).fail(function() {
-      // TODO: some error state.
-      console.log('Card list retrieval failed');
-    });
+    var user = UserStore.getLoggedInUser();
```

Comment:
The singletons are easy to mock which is nice. I looked at some flux libs but I decided I wanted to try a roll your own approach first just to see how annoying it was. I think we can simplify it further for sure. This article kind of convinced me that there aren't good libs yet https://medium.com/@dan_abramov/the-evolution-of-flux-frameworks-6c16ad26bb31 Redux was one I looked at it but it doesn't seem production ready yet.